### PR TITLE
[DOCS] Theme selector

### PR DESCRIPTION
**Pull request by deborahniesz**
[Deploy Preview](https://deploy-preview-10181.docs.greatexpectations.io/)

[Mockups](https://xd.adobe.com/view/9ef7232c-80ce-4ad7-9c3f-1a96126f84b4-cba4/screen/5a1c1da9-4732-49fe-bc9a-0312917fee86/)

[Jira Ticket](https://greatexpectations.atlassian.net/jira/software/c/projects/DSB/boards/79?issueParent=26175%2C27414&selectedIssue=DSB-656)

https://github.com/user-attachments/assets/e6dc0338-39ee-4910-a40c-309324c03f40

EDIT: Also removed the opacity in the navbar (both in desktop at the top side of the screen and in mobile with the sidebar menu), this change applies to the dark AND light theme.

image

image

image

image

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: docs/docusaurus/src/components/WasThisHelpful/styles.module.scss [hunk 1]
```diff
@@ -216,3 +216,40 @@
         max-height: unset;
     }
 }
+
+[data-theme='dark'] {
+    .feedbackCard {
+        background-color: #404041;
+        border-color: #404041;
+        box-shadow: 0 3px 6px #00000029;
```
Comment (josectobar) [inline docs/docusaurus/src/components/WasThisHelpful/styles.module.scss]: Can we make these variables? I see we use them in a couple of places.

File: docs/docusaurus/src/css/custom.scss [hunk 2]
```diff
@@ -296,3 +305,15 @@ li p + ol {
 .code-block-hide-line {
   display:none;
 }
+
+:root[data-theme='dark'] {
+  --ifm-font-color-base: #FAFAFA;
+  --ifm-navbar-link-color: #FAFAFA;
+  --ifm-menu-color: #FAFAFA;
+  --ifm-background-color: #262626;
```
Comment (josectobar) [inline docs/docusaurus/src/css/custom.scss]: Can we add these hex to their  own var and then just reference them here instead of repeating them?

**Comment by josectobar** [comment]
It's not related to these changes, but we need to fix the font color of the privacy link in light mode. It seems to be hidden.

**Comment by josectobar** [comment]
Could we remove the white borders in the code snippet boxes for dark mode? 

image

**Comment by josectobar** [comment]
some embeded span texts are shown with a white background, can we remove the background and the border?

before: 
image


after:

image

**Comment by josectobar** [comment]
We should also change the white background and text color from docs/reference/api/core/util code snippets:

before: 

image


after:

image

**Comment by josectobar** [comment]
The disabled state of the submit button in the feedback modal lacks enough contrast: 
image

**Comment by josectobar** [comment]
There are some screenshots with white backgrounds. Are we planning on replacing these at some point? 
image

**Review by josectobar — CHANGES_REQUESTED** [review]
Great work! I left a couple of comments related to the code and some others related to specific cases where I think we need to adjust some of the styling for dark mode. cc. @SuperYeehee

**Comment by SuperYeehee** [comment]
> Great work! I left a couple of comments related to the code and some others related to specific cases where I think we need to adjust some of the styling for dark mode. cc. @SuperYeehee

Could you clarify what you mean on "Adjust some of the styling for DM?" Are you references like a set of standard variables like we use in our GX theme.ts?

**Comment by josectobar** [comment]
> > Great work! I left a couple of comments related to the code and some others related to specific cases where I think we need to adjust some of the styling for dark mode. cc. @SuperYeehee
> 
> Could you clarify what you mean on "Adjust some of the styling for DM?" Are you references like a set of standard variables like we use in our GX theme.ts?

@SuperYeehee , I'm referring to my comments above☝🏽

File: docs/docusaurus/src/css/api_docs/api_docs.scss [hunk 3]
```diff
@@ -34,7 +34,26 @@
...
+          border: none;
+          color: var(--ifm-font-color-base);
+        }
+
+        pre {
+          border: none;
+        }
+
+       .highlight pre {
+         background-color: #404041;
```
Comment (josectobar) [inline docs/docusaurus/src/css/api_docs/api_docs.scss]:
```suggestion
         background-color: var(--payne-gray-color);
```

File: docs/docusaurus/src/css/api_docs/pydata-sphinx-theme-modified.scss [hunk 4]
```diff
@@ -10288,27 +10290,49 @@ nav.page-toc ul > li {
 }
 
 div.deprecated, div.versionadded, div.versionchanged {
-  background-color: var(--pst-color-on-background);
-  border-left: .2rem solid;
-  border-color: var(--pst-color-info);
+  border: 1px solid;
   border-radius: .25rem;
-  box-shadow: 0 .2rem .5rem var(--pst-color-shadow), 0 0 .0625rem var(--pst-color-shadow) !important;
+  box-shadow: none !important;
```
Comment (josectobar) [inline docs/docusaurus/src/css/api_docs/pydata-sphinx-theme-modified.scss]: Does this work without the `!important`?

File: docs/docusaurus/src/css/api_docs/pydata-sphinx-theme-modified.scss [hunk 5]
```diff
@@ -10336,11 +10367,18 @@ div.versionchanged p:before {
 }
 
 div.deprecated {
-  border-color: var(--pst-color-danger)
-}
+  background-color: #FEF6F9;
+  border: 1px solid #E54F54;
```
Comment (josectobar) [inline docs/docusaurus/src/css/api_docs/pydata-sphinx-theme-modified.scss]:
```suggestion
  border: 1px solid var(--ifm-color-warning-dark);
```

File: docs/docusaurus/src/css/api_docs/pydata-sphinx-theme-modified.scss [hunk 6]
```diff
@@ -10336,11 +10367,18 @@ div.versionchanged p:before {
...
 
-div.deprecated p:before {
-  background-color: var(--pst-color-danger)
+  .deprecated:before {
+    mask: url("/static/img/admonition-info-icon.svg");
+  }
+
+  cite {
+    background-color: #FEF6F9;
+    border-color: #E54F54;
```
Comment (josectobar) [inline docs/docusaurus/src/css/api_docs/pydata-sphinx-theme-modified.scss]:
```suggestion
    border-color: var(--ifm-color-warning-dark:);
```

File: docs/docusaurus/src/css/api_docs/pydata-sphinx-theme-modified.scss [hunk 7]
```diff
@@ -10320,11 +10344,18 @@ div.deprecated > p:before, div.versionadded > p:before, div.versionchanged > p:b
...
-  border-color: var(--pst-color-success)
-}
+  background-color: #F8F9EC;
+  border: 1px solid #B6C647;
+  border-left-width: 6px;
 
-div.versionadded p:before {
-  background-color: var(--pst-color-success)
+  cite {
+    background-color: #b6c64733;
```
Comment (josectobar) [inline docs/docusaurus/src/css/api_docs/pydata-sphinx-theme-modified.scss]:
```suggestion
    background-color: var(--ifm-color-success-dark);
```

File: docs/docusaurus/src/css/api_docs/pydata-sphinx-theme-modified.scss [hunk 8]
```diff
@@ -10320,11 +10344,18 @@ div.deprecated > p:before, div.versionadded > p:before, div.versionchanged > p:b
...
+    border-color: #b6c647;
```
Comment (josectobar) [inline docs/docusaurus/src/css/api_docs/pydata-sphinx-theme-modified.scss]:
```suggestion
    border-color: var(--ifm-color-success-dark);
```

File: docs/docusaurus/src/css/api_docs/pydata-sphinx-theme-modified.scss [hunk 9]
```diff
@@ -10320,11 +10344,18 @@ div.deprecated > p:before, div.versionadded > p:before, div.versionchanged > p:b
 }
 
 div.versionadded {
-  border-color: var(--pst-color-success)
-}
+  background-color: #F8F9EC;
+  border: 1px solid #B6C647;
```
Comment (josectobar) [inline docs/docusaurus/src/css/api_docs/pydata-sphinx-theme-modified.scss]:
```suggestion
  border: 1px solid var(--ifm-color-success-dark);
```

**Review by josectobar — APPROVED** [review]
Looks great! I love dark mode! please address some final suggestions but otherwise is good to go.

**Pull request merged**